### PR TITLE
 [CWS] re-add register to iterator

### PR DESCRIPTION
**Pull request by safchain**
### What does this PR do?

This allows writing rule targeting the same level of process ancestors : 

```
open.file.path == "/tmp/tes" && process.ancestors[A].name == "bash" && process.ancestors[A].uid != 0
```

### CODE REVIEW

**Review by maycmlee — APPROVED** [review]

File: pkg/security/secl/compiler/eval/model_test.go [hunk 1]
```diff
@@ -209,8 +224,44 @@ func (m *testModel) GetEvaluator(field Field, _ RegisterID) (Evaluator, error) {
...
+				return ctx.Event.(*testEvent).process.list.Len()
+			},
+			Field: field,
+		}, nil
+
 	case "process.list.key":
 
+		if regID != "" {
+
+			fmt.Printf("ZAAAA: %s\n", regID)
```
Comment (spikat) [inline pkg/security/secl/compiler/eval/model_test.go:239]: ZAAAA ? :)

File: pkg/security/secl/compiler/eval/model_test.go [hunk 2]
```diff
@@ -209,8 +224,44 @@ func (m *testModel) GetEvaluator(field Field, _ RegisterID) (Evaluator, error) {
...
+
+			return &IntArrayEvaluator{
+				EvalFnc: func(ctx *Context) []int {
+					idx := ctx.Registers[regID]
+
+					var i int
+
+					fmt.Printf("ZOOOOOO: %s => %d\n", regID, idx)
```
Comment (spikat) [inline pkg/security/secl/compiler/eval/model_test.go:247]: Even ZOOOO ?? :)

File: pkg/security/secl/compiler/eval/model_test.go [hunk 3]
```diff
@@ -232,6 +283,32 @@ func (m *testModel) GetEvaluator(field Field, _ RegisterID) (Evaluator, error) {
...
 	case "process.list.value":
 
+		if regID != "" {
+			return &StringArrayEvaluator{
+				EvalFnc: func(ctx *Context) []string {
+					idx := ctx.Registers[regID]
+
+					var i int
+
+					fmt.Printf("ZOOOOOO: %s => %d\n", regID, idx)
```
Comment (spikat) [inline pkg/security/secl/compiler/eval/model_test.go:293]: Here too

File: pkg/security/secl/compiler/eval/model_test.go [hunk 4]
```diff
@@ -232,6 +283,32 @@ func (m *testModel) GetEvaluator(field Field, _ RegisterID) (Evaluator, error) {
...
+
+					el := ctx.Event.(*testEvent).process.list.Front()
+					for el != nil {
+						if i == idx {
+							fmt.Printf("GGGGG: %s => %d\n", regID, idx)
```
Comment (spikat) [inline pkg/security/secl/compiler/eval/model_test.go:298]: also

File: pkg/security/tests/process_test.go [hunk 5]
```diff
@@ -141,15 +141,15 @@ func TestProcessContext(t *testing.T) {
...
+			Expression: `open.file.path == "{{.Root}}/test-process-ancestors" && process.ancestors[A].file.name in ["sh", "dash", "bash"]`,
 		},
 		{
 			ID:         "test_rule_parent",
 			Expression: `open.file.path == "{{.Root}}/test-process-parent" && process.parent.file.name in ["sh", "dash", "bash"]`,
 		},
 		{
 			ID:         "test_rule_pid1",
-			Expression: `open.file.path == "{{.Root}}/test-process-pid1" && process.ancestors[_].pid == 1`,
+			Expression: `open.file.path == "{{.Root}}/test-process-pid1" && process.ancestors.pid == 1`,
```
Comment (spikat) [inline pkg/security/tests/process_test.go:152]: Will be more "logic" to put back [A]

**Review by spikat — APPROVED** [review]

**Pull request merged**